src/entt/entity/polymorphic.hpp

```diff
@@ -0,0 +1,554 @@
+
+    bool operator==(const poly_components_iterator& other) const ENTT_NOEXCEPT {
+        return it == other.it;
+    }
+
+    bool operator!=(const poly_components_iterator& other) const ENTT_NOEXCEPT {
+        return it != other.it;
+    }
+
+    pointer operator->() ENTT_NOEXCEPT {
+        return current;
+    }
+
+    reference operator*() ENTT_NOEXCEPT {
+        if constexpr(std::is_pointer_v<value_type>) {
+            return operator->();
+        } else {
+            return *operator->();
+        }
+    }
+
+private:
+    entity_type ent;
+    PoolsIterator it;
+    PoolsIterator end;
+    pointer current = nullptr;
+};
+
+} // namespace entt::internal
+
+/**
+ * Internal details not to be documented.
+ * @endcond
+ */
+
+
+namespace entt {
+
+/** @brief base class for entt::poly_pool_holder */
+template<typename Entity>
+class poly_pool_holder_base {
+public:
+    inline poly_pool_holder_base(basic_sparse_set<Entity>* pool,
+                                 void* (*getter)(void*, Entity) ENTT_NOEXCEPT,
+                                 bool (*remover)(void*, Entity)) :
+          pool_ptr(pool), getter_ptr(getter), remover_ptr(remover) {}
+
+protected:
+    basic_sparse_set<Entity>* pool_ptr;
+    void* (*getter_ptr)(void*, Entity) ENTT_NOEXCEPT;
+    bool (*remover_ptr)(void*, Entity);
+};
+
+/**
+ * @brief Holds pointer to child type storage and a function to convert it into parent type
+ * @tparam Entity underlying entity type
+ * @tparam Type polymorphic component type to be converted into
+ */
+template<typename Entity, typename Type>
+class poly_pool_holder : public poly_pool_holder_base<Entity> {
+public:
+    using poly_pool_holder_base<Entity>::poly_pool_holder_base;
+
+    /** @brief entity type of the component pools */
+    using entity_type = Entity;
+    /** @brief component type */
+    using value_type = Type;
+    /** @brief pointer to the component, will be same as value_type, if component itself is a pointer */
+    using pointer_type = std::remove_pointer_t<Type>*;
+    /** @copydoc pointer_type */
+    using const_pointer_type = const std::remove_pointer_t<Type>*;
+
+    /**
+     * @brief gets child from a child pool as parent type reference by given entity
+     * @param ptr pointer from the child type set
+     * @return Returns reference to Type, converted from a given pointer. Will return pointer instead of reference, if parent type is a pointer.
+     */
+    inline pointer_type try_get(const Entity ent) ENTT_NOEXCEPT {
+        return pointer_type(this->getter_ptr(this->pool_ptr, ent));
+    }
+
+    /** @copydoc try_get */
+    inline const_pointer_type try_get(const Entity ent) const ENTT_NOEXCEPT {
+        return const_cast<poly_pool_holder<Entity, Type>*>(this)->try_get(ent);
+    }
+
+    /**
+     * @brief removes component from pool by entity
+     * @param ent entity
+     * @return true, if component was removed, false, if it didnt exist
+     */
+    inline bool remove(const Entity ent) {
```

Comment:
I can't find where you _fill_ the `remover_ptr` but I still feel like you don't really need to.
You can delete an entity from a pointer to a sparse set. No need to wrap that functionality probably?
That is, the following should work just fine afaik:
```cpp
pool_ptr->remove(ent); // or erase too, not sure what the purpose is here
```